packages/map-template/src/components/MapTemplate/MapTemplate.jsx

```diff
@@ -686,6 +689,16 @@ function MapTemplate({ apiKey, gmApiKey, mapboxAccessToken, venue, locationId, p
         setSelectedCategory(null); // unselect category when route is finished
     }
 
+    /**
+     * Function that handles the finishing of route.
+     */
+    function onRouteFinish() {
+        finishRoute();
+        resetAppHistory();
+        setResetCount(curr => curr + 1);
```

Comment:
I think it deserves the same explanation in a comment as in `resetStateAndUI`, is it is a somewhat controversial thing to do.